File: Battleships-Angular/src/app/register-user/PlayerRegisterModel.ts
export class PlayerRegisterModel {

  constructor(
    email: string,
    firstName: string,
    lastName: string,
    nickName: string,
    password: string,
    confirmPassword: string
  ) {
    this.Email = email;
    this.FirstName = firstName;
    this.LastName = lastName;
    this.NickName = nickName;
    this.Password = password;
    this.ConfirmPassword = confirmPassword;
  }

  Email: string;
  FirstName: string;
  LastName: string;
  NickName: string;
  Password: string;
  ConfirmPassword: string;
}

File: Battleships.BLL/Models/PlayerRegisterModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Battleships.BLL
{
    public class PlayerRegisterModel
    {
        public PlayerRegisterModel()
        {

        }

        [Required]
        [EmailAddress]
        [DataType(DataType.EmailAddress)]
        public string Email { get; set; }

        [Required]
        public string FirstName { get; set; }

        public string LastName { get; set; }

        [Required]
        public string NickName { get; set; }

        [Required]
        [MinLength(7)]
        [MaxLength(15)]
        [DataType(DataType.Password)]
        public string Password { get; set; }

        [Required]
        [MinLength(7)]
        [MaxLength(15)]
        [DataType(DataType.Password)]
        public string ConfirmPassword { get; set; }
    }
}
